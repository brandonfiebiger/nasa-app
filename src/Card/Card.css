.Card {
  height: 30rem;
  width: 20rem;
  border: 1px solid black;
  margin-left: 2rem;
  margin-top: 1rem;
}

img {
  height: 20rem;
  width: 20rem;
}