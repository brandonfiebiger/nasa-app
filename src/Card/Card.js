import React, { Component } from 'react';

export class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      toggled: false
    }
  }


  render() {
    const { image, date, id, rover, camera } = this.props.photo;
    return(
      <div>
        <img src={image} />
      </div>
    )
  }
}