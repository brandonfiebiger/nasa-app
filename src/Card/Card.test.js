import { shallow } from 'enzyme';
import React from 'react';
import Card from './Card';


describe('Card', () => {

  let wrapper;

  let mockPhoto = {
    image: 'image',
    date: 'date',
    rover: 'rover',
    camera: 'camera',
    id: 1
  }

  beforeEach(() => {
    wrapper = shallow(<Card photo={mockPhoto} />)
  })


  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should change its toggled state when handleToggle is called', () => {
    expect(wrapper.state('toggled')).toEqual(false);
    wrapper.instance().handleToggle();
    expect(wrapper.state('toggled')).toEqual(true);
  })
})