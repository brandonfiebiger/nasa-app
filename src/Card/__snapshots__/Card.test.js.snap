// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    photo={
      Object {
        "camera": "camera",
        "date": "date",
        "id": 1,
        "image": "image",
        "rover": "rover",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="image"
      />,
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "image",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="image"
        />,
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "image",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
